lib/aasm/persistence/active_record_persistence.rb

```diff
@@ -76,7 +76,12 @@ def aasm_write_state(state, name=:default)
             self.save
           end
 
-          success ? true : aasm_rollback(name, old_value)
+          unless success
+            aasm_rollback(name, old_value)
+            raise ActiveRecord::RecordInvalid.new(self) if aasm_whiny_persistence(name)
+          end
+
+          success
```

Comment:
Since the only possible value for `success` at this point is `true`, I'd be explicit and would return `true` here instead of `success` .
Might be a personal thing, though.